content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,537 @@
+   return 1
+ else:
+   return 2
+ 
+df["species"] = df.apply(species_cat, axis=1)
+```
+```python
+df.sample(5)
+```
+![dfsample5](/engineering-education/statistical-hypothesis-testing-python-implementation/dfsample5.PNG)
+```python
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+```
+```python
+df_new.head()
+```
+![newhead](/engineering-education/statistical-hypothesis-testing-python-implementation/df_new.PNG)
+ 
+Let's make use of `chic2_contigency` provided for us in the `scipy.stats` library.
+```python
+stat, p, dof, expected= chi2_contingency(df_new)
+ 
+print(dof)
+```
+```bash
+The degree of freedom is: 149
+```
+We can comfortably get 3 outputs in test statistics, degree of freedom, p-value, and expected values.
+ 
+[Test statistic](https://online.stat.psu.edu/statprogram/reviews/statistical-concepts/chi-square-tests) is a characterized feature of significance in the Chi-square test that helps in determining how uncommon the result might be provided that null hypothesis is assumed to be true.
+ 
+The number of features or variables that we have access to varies by the degree of freedom. Follow the link for more information on [degree of freedom](https://byjus.com/maths/chi-square-test/).
+ 
+Hence, we set our probability of confidence to 95% as explained earlier in the tutorial.
+```python
+prob = 0.95
+ 
+alpha = 1.0 - prob
+ 
+print('The alpha/significance level=%.3f' % alpha)
+print('The P-value is =%.2f' % p)
+ 
+if p <= alpha:
+ print('Reject the Null Hypothesis ( reject H0)')
+else:
+ print('Accept the Null Hypothesis (fail to reject H0)')
+```
+```bash
+alpha/significance level=0.050
+P-value =0.10
+Accept the Null Hypothesis (fail to reject H0)
+```
+The `p-value` is greater than the `alpha value` set at 0.05, we will fail to reject the null hypothesis and accept the null hypothesis. This means the petal width and species of flower are not dependent, i.e. there is no relationship between them.
+ 
+### T-Test - Python implementation
+ 
+The T-test is utilized when we plan to evaluate the discrepancy between the means of two groups of samples. Unlike the Chi-Square Test, the T-test is used on continuous variables.
+ 
+T-tests has three basic types
+ 
+1. **One sample t-test:** It checks whether the population has a different mean from the sample.
+The mathematical formula is:
+ 
+[t-test = a-BX∕N](https://www.datanovia.com/en/lessons/t-test-formula/one-sample-t-test-formula/)
+ 
+Where
+ 
+the observed mean group is a
+ 
+the assumed population means is B
+ 
+the standard deviation of the data group is X
+ 
+the number of observations in the group is V
+ 
+2. **Two-sample t-test:** It checks and compares the means of two groups that are not dependent on each other and compares the population's means to see if there is a huge difference.
+The mathematical expression is:
+ 
+[t-test = Ma – (Mb*v*Na)+vNb](https://www.statology.org/two-sample-t-test/)
```

Comment:
Please remove the link from the formula.
Also, I've mentioned this before, please attach images rather than writing the formulae, like how you did the chi-square test formula. Please do this for the others as well.